tests/model/test_model.py

```diff
@@ -2079,20 +2101,60 @@ def test_toggle_stream_pinned_status(
                     "id": 0,
                 },
                 True,
+                False,
+                False,
+                id="op_stop",
             ),
         ],
-        ids=["not_in_pm_narrow", "not_in_pm_narrow_with_sender", "start", "stop"],
     )
-    def test__handle_typing_event(self, mocker, model, narrow, event, called):
+    def test__handle_typing_event(
+        self,
+        mocker,
+        model,
+        narrow,
+        event,
+        is_animation_in_progress,
+        expected_is_recipient_typing,
+        expected__show_typing_animation,
+    ):
         event["type"] = "typing"
 
-        mocker.patch("zulipterminal.ui.View.set_footer_text")
+        mocker.patch("zulipterminal.model.Model._show_typing_animation")
         model.narrow = narrow
         model.user_dict = {"hamlet@zulip.com": {"full_name": "hamlet"}}
+        model.is_recipient_typing = False
+        model.is_typing_animation_in_progress = is_animation_in_progress
 
         model._handle_typing_event(event)
 
-        assert model.controller.view.set_footer_text.called == called
+        assert model.is_recipient_typing == expected_is_recipient_typing
+        assert model._show_typing_animation.called == expected__show_typing_animation
+
+    @pytest.mark.parametrize(
+        "event, expect_set_footer_animation",
```

Comment:
minor: expected*

Reply:
I think expect is a better word here. Expected would we apt when we compare a variable with its expected value, but here every possibility makes a call to set_footer_text so I've used the above which would translate to - `if you expect footer animation to be set then do so and so tests else do something else`

Reply:
Maybe `expect_footer_animation` would be better?

Reply:
Alternatively, you could call it `is_set_footer_animation_called`?